applications/StructuralMechanicsApplication/custom_elements/membrane_element.cpp

```diff
@@ -0,0 +1,1292 @@
+                point_number);
+
+            if (rVariable==PRINCIPLE_PK2_STRESS_VECTOR){
+                Vector principle_stresses = ZeroVector(2);
+                PrincipleVector(principle_stresses,stress);
+                rOutput[point_number] = principle_stresses;
+            }  else {
+                rOutput[point_number] = stress;
+            }
+        }
+
+    }  else if (rVariable==CAUCHY_STRESS_VECTOR || rVariable==PRINCIPLE_CAUCHY_STRESS_VECTOR){
+
+        Vector stress = ZeroVector(3);
+        array_1d<Vector,2> current_covariant_base_vectors;
+        array_1d<Vector,2> reference_covariant_base_vectors;
+        array_1d<Vector,2> reference_contravariant_base_vectors;
+
+        array_1d<Vector,2> transformed_base_vectors;
+
+        Matrix covariant_metric_current = ZeroMatrix(3);
+        Matrix covariant_metric_reference = ZeroMatrix(3);
+        Matrix contravariant_metric_reference = ZeroMatrix(3);
+        Matrix inplane_transformation_matrix_material = ZeroMatrix(3);
+
+        Matrix stress_matrix = ZeroMatrix(2);
+        Matrix deformation_gradient = ZeroMatrix(2);
+        double det_deformation_gradient = 0.0;
+
+        for (SizeType point_number = 0; point_number < r_integration_points.size(); ++point_number){
+            // getting information for integration
+            const Matrix& shape_functions_gradients_i = r_shape_functions_gradients[point_number];
+
+            CovariantBaseVectors(current_covariant_base_vectors,shape_functions_gradients_i,ConfigurationType::Current);
+            CovariantBaseVectors(reference_covariant_base_vectors,shape_functions_gradients_i,ConfigurationType::Reference);
+
+            CovariantMetric(covariant_metric_current,current_covariant_base_vectors);
+            CovariantMetric(covariant_metric_reference,reference_covariant_base_vectors);
+            ContravariantMetric(contravariant_metric_reference,covariant_metric_reference);
+
+            ContraVariantBaseVectors(reference_contravariant_base_vectors,contravariant_metric_reference,reference_covariant_base_vectors);
+
+            TransformBaseVectors(transformed_base_vectors,reference_contravariant_base_vectors);
+
+            InPlaneTransformationMatrix(inplane_transformation_matrix_material,transformed_base_vectors,reference_contravariant_base_vectors);
+
+            StressPk2(stress,contravariant_metric_reference,covariant_metric_reference,
+                covariant_metric_current,transformed_base_vectors,inplane_transformation_matrix_material,
+                point_number);
+
+            DeformationGradient(deformation_gradient,det_deformation_gradient,current_covariant_base_vectors,reference_contravariant_base_vectors);
+
+            stress_matrix(0,0) = stress[0];
+            stress_matrix(1,1) = stress[1];
+            stress_matrix(0,1) = stress[2];
+            stress_matrix(1,0) = stress[2];
+
+            stress_matrix = prod(deformation_gradient,stress_matrix);
```

Comment:
ouch this use is really painful. it is correct since you assume a temporary is created in each line, but for example it would break with AMatrix. do allocate a temp...